File: Easy/70-climbing-stairs/try-3.cpp
/*
 * Problem #70: Climbing Stairs
 * Difficulty: Easy
 * Submission: Try 3
 * status: Accepted
 * Language: cpp
 * Date: 22/03/2026, 16:21:58
 * Link: https://leetcode.com/problems/climbing-stairs/
 */

class Solution {
public:
    unordered_map<int, int> dp;
    int climbStairs(int n) {
        if(n <= 1) return 1;
        int a = 1, b = 1;
        for(int i = 2;i <= n; i++){
            int next = a + b;
            b = a;
            a = next;
        }
        return a;
    }
};

File: Easy/1013-partition-array-into-three-parts-with-equal-sum/try-3.py
/*
 * Problem #1013: Partition Array Into Three Parts With Equal Sum
 * Difficulty: Easy
 * Submission: Try 3
 * status: Accepted
 * Language: python3
 * Date: 20/12/2024, 15:13:37
 * Link: https://leetcode.com/problems/partition-array-into-three-parts-with-equal-sum/
 */


class Solution:
    def canThreePartsEqualSum(self, arr: List[int]) -> bool:
        total = sum(arr)
        c=0
        if total % 3 != 0:
            return False
        
        per_arr = total // 3
        part = 0
        for i in arr:
            part += i
            if part == per_arr:
                c+=1
                part=0
        
        return c>=3 
        



        
